### PR TITLE
dev: add testing section and move smoke tests to test/smoke

### PR DESCRIPTION
**Pull request by mastercactapus**
**Description:**
This PR moves `smoketest` to `test/smoke` with the relevant import and package name updates.
Additionally, a `Testing` section was added to CONTRIBUTING.md outlining where and how we use the different types of tests in GoAlert.

**Which issue(s) this PR fixes:**
Part of #2589

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -35,5 +35,15 @@ Be sure to run `make check` and tests before opening a PR to catch common errors
...
+
+## Testing
+
+GoAlert utilizes 3 main types of testing as tools for different purposes:
+
+- Unit tests are used for complicated logic and exhaustive edge-case testing and benchmarking. They live with the code being tested:
+  - For backend code, a `_test.go` version of a file will contain relevant unit tests. More info [here](https://pkg.go.dev/testing)
+  - For UI code, a `.test.ts` version of a file will contain relevant unit tests. More info [here](https://jestjs.io/docs/getting-started).
+- Smoke tests (in `test/smoke`) are used to ensure main functionality and things like behavioral compatibility with future versions & DB migrations. These focus on hard guarantees like deliverability and preserving intent as the application and datastore evolves and changes over time.
+- Integration tests (currently under `web/src/cypress/integration`) are primarily used to validate happy-path flows work end-to-end, and any important/common error scenarios. They are focused on UX and high-level funcationality.
```
Comment (Forfold) [inline CONTRIBUTING.md:49]:
```suggestion
- Integration tests (currently under `web/src/cypress/integration`) are primarily used to validate happy-path flows work end-to-end, and any important/common error scenarios. They are focused on UX and high-level functionality.
```

**Review by Forfold — APPROVED** [review]

**Review by KatieMSB — APPROVED** [review]
